File: morning_exercises/day6.cs
public static class FindMaxProfit
{
    public static int MaxProfit(List<int> prices)
    {
        int minPrice = int.MaxValue, maxProfit = 0;

        foreach (int price in prices)
        {
            minPrice = Math.Min(minPrice, price);
            maxProfit = Math.Max(maxProfit, price - minPrice);
        }
        return maxProfit;
    }
}

File: morning_exercises/day6.ts
function maxProfit(prices: number[]): number {
    let minPrice = Number.MAX_VALUE, maxProfit = 0;

    prices.forEach(price => {
        minPrice = Math.min(minPrice, price);
        maxProfit = Math.max(maxProfit, price - minPrice);
    });
    
    return maxProfit;
}

console.log(maxProfit([1,2,3,4,5]));